lib/src/main/kotlin/com/lemonappdev/konsist/api/ext/list/KoValueProviderListExt.kt

```diff
@@ -5,21 +5,42 @@ import com.lemonappdev.konsist.api.provider.KoValueProvider
 /**
  * List containing elements with value.
  *
- * @param value The value to include.
  * @param values The value(s) to include.
- * @return A list containing elements with the specified values.
+ * @return A list containing elements with the specified values (or any value if [values] is empty).
  */
-fun <T : KoValueProvider> List<T>.withValue(value: String, vararg values: String): List<T> = filter {
-    it.value == value || values.any { value -> it.value == value }
+fun <T : KoValueProvider> List<T>.withValue(vararg values: String): List<T> = filter {
+    when {
+        values.isEmpty() -> it.hasValue()
+        else -> values.any { value -> it.hasValue(value) }
+    }
 }
 
 /**
  * List containing elements without value.
  *
- * @param value The value to exclude.
  * @param values The value(s) to exclude.
- * @return A list containing elements without the specified values.
+ * @return A list containing elements without the specified values (or none value if [values] is empty).
  */
-fun <T : KoValueProvider> List<T>.withoutValue(value: String, vararg values: String): List<T> = filter {
-    it.value != value && values.none { value -> it.value == value }
+fun <T : KoValueProvider> List<T>.withoutValue(vararg values: String): List<T> = filter {
+    when {
+        values.isEmpty() -> !it.hasValue()
+        else -> values.none { value -> it.hasValue(value) }
+    }
 }
+
```

Comment:
`value` has type `String?`
The predicate should take `String?` or `String`?

Reply:
with value should be allowed on `String?` because the value can be nullable (I think it would crash for `String`).

However, I believe that `?: false` can be hidden inside `withoutValue` function ( `?: true` can be hidden inside `withValue` function) - after evaluating the expression (passed as an argument) we can check if it is null and return this default value.

Reply:
Odpisze po polsku, ale będzie mi łatwiej opisać problem:

Wydaje mi się, że teraz działa tak, jak powinno, czyli `predicate` przyjmuje `String`, ale jest ok, bo tylko na wartościach `non-null` jest wywoływany.

Rozważam jeszcze dodanie parametru `strict` który by dawał możliwość określenia czy chcemy żeby nullowalne wartości zostały zwracane w liście na koniec. No bo w przypadku takiego testu:
```kotlin
@Test
    fun `withValue{predicate} returns declaration with value matching to predicate`() {
        // given
        val value1 = "sampleValue1"
        val value2 = "sampleValue2"
        val prefix = "sample"
        val suffix = "1"
        val predicate: (String) -> Boolean = { it.startsWith(prefix) && it.endsWith(suffix) }
        val declaration1: KoValueProvider = mockk {
            every { value } returns value1
        }
        val declaration2: KoValueProvider = mockk {
            every { value } returns value2
        }
        val declaration3: KoValueProvider = mockk {
            every { value } returns null
        }
        val declarations = listOf(declaration1, declaration2, declaration3)

        // when
        val sut = declarations.withValue(predicate)

        // then
        sut shouldBeEqualTo listOf(declaration1, declaration3)
    }
```
nie jestem przekonana, że chcemy mieć `declaration3` w `sut`. 

Więc albo dodamy `strict` i wtedy będzie można okreslić czego się oczekuje albo bardziej bym poszła w stronę, że te extensions mają w sobie założenie że ma `value` inną niż null i ponadto spełnia warunek